libimage/pull.go

```diff
@@ -379,6 +381,48 @@ func (r *Runtime) copyFromRegistry(ctx context.Context, ref types.ImageReference
 		return nil, err
 	}
 
+	// Avoid running out of time when running inside a systemd unit by
+	// regularly increasing the timeout.
+	if socketPath, ok := os.LookupEnv("NOTIFY_SOCKET"); ok {
+		socketAddr := &net.UnixAddr{
+			Name: socketPath,
+			Net:  "unixgram",
+		}
+		conn, err := net.DialUnix(socketAddr.Net, nil, socketAddr)
+		if err != nil {
+			return nil, err
+		}
+		defer conn.Close()
+
+		timeout := 30 * time.Second
```

Comment:
another thing that must be tested because I assume the current version will be racy
If we extend by 30 seconds and wait 30 seconds then I think there is a race were systemd could run into the 30 timeout before the podman process send the next EXTEND_TIMEOUT_USEC message, this would depend on the scheduler.

So I think the safe way is to extend by 30s but write a new message every 25s or something like that.
Because the docs make this very clear:
> The value specified is a time in microseconds during which the service must send a new message.

Reply:
I also thought about this issue. From the name I thought it is an extension to an already existing timeout. And if so, the yes, there is a race that decreases the overall timeout but shouldn't become an issue.
Having said that, if this value becomes the new timeout (i.e. not extending the current timeout value), I completely agree and we should wait less then the time we extend the timeout.

Reply:
I think the docs are very clear the new message must be send within the timeout specified.